test/421-prefetch-fill-level.cc

```diff
@@ -72,3 +72,97 @@ SCENARIO("A prefetch can be issued without creating an MSHR") {
   }
 }
 
+SCENARIO("A prefetch fill the first level") {
+  GIVEN("An empty cache") {
+    constexpr uint64_t hit_latency = 1;
+    constexpr uint64_t fill_latency = 10;
+    do_nothing_MRC mock_ll;
+    CACHE::NonTranslatingQueues uut_queues{1, 32, 32, 32, 0, hit_latency, LOG2_BLOCK_SIZE, false};
+    CACHE uut{"421-uut", 1, 1, 8, 32, fill_latency, 1, 1, 0, false, false, false, (1<<LOAD)|(1<<PREFETCH), uut_queues, &mock_ll, CACHE::pprefetcherDno, CACHE::rreplacementDlru};
+
+    // Initialize the prefetching and replacement
+    uut.impl_prefetcher_initialize();
+    uut.impl_replacement_initialize();
+
+    // Turn off warmup
+    uut.warmup = false;
+    uut_queues.warmup = false;
+    uut.begin_phase();
+    uut_queues.begin_phase();
+
+    WHEN("A prefetch is issued with 'fill_this_level == true'") {
+      auto seed_result = uut.prefetch_line(0xdeadbeef, true, 0);
+      REQUIRE(seed_result);
+
+      for (int i = 0; i < 1000; i++) {
+        uut_queues._operate();
+        uut._operate();
+        mock_ll._operate();
+      }
+
+      // We ensure that the block fill the first level
+      bool find = false;
+      for (auto i: uut.block) if (find = (i.address == 0xdeadbeef)) break;
```

Comment:
Could you please perform this check by using the cache's interfaces? See [a similar line in test 420](https://github.com/ChampSim/ChampSim/blob/2aa99f98d69f3467d765141c2bd31e2b474ca256/test/420-issue-prefetch.cc#L50).

Performing tests via the interfaces checks for correct behavior, rather than correct implementation. Specifically, we are testing that a prefetch issued with `fill_this_level == true` causes a later hit in the second level, but not in the first level. So, we want to test that the return time of a packet with the same address has the miss latency of the first level, but not the miss latency of the second level. If we do it this way, we're free to change the implementation of the cache without breaking the tests.

Reply:
I see, I change the test to be at the same style like the 420.